src/pages/Main/Components/MainSection/MainSection.js

```diff
@@ -5,44 +5,85 @@ export default class MainSection extends React.Component {
   constructor() {
     super();
     this.state = {
-      movieInfomationList: [],
+      width: '',
+      index: 0,
     };
+    this.containerWidth = React.createRef();
   }
-  componentDidMount() {
-    fetch('http://localhost:3000/data/MainMockData.json', {
-      method: 'GET',
-    })
-      .then(res => res.json())
-      .then(movieInfomationList => {
-        this.setState({
-          movieInfomationList,
-        });
-      });
-  }
+
+  //다음 슬라이드 이동 함수
+  handleNextSliding = () => {
+    const { index } = this.state;
+    const { movieInformationList } = this.props;
+    if (index === Math.floor(movieInformationList.length / 5) - 1) {
+      return;
+    }
+    this.setState({
+      index: index + 1,
+      width: this.containerWidth.current.scrollWidth * (index + 1),
+    });
+  };
+
+  //이전 슬라이드 이동 함수
+  handlePreSliding = () => {
+    const { index } = this.state;
+    if (index === 0) {
+      return;
+    }
+    this.setState({
+      index: index - 1,
+      width: this.containerWidth.current.scrollWidth * (index - 1),
+    });
+  };
+
   render() {
-    const { movieInfomationList } = this.state;
+    const { width, index, movieListRanking } = this.state;
+    const { listTitle, movieInformationList } = this.props;
+    console.log('안덱스값', movieInformationList);
     return (
       <section className="mainSection">
         <div className="mainTitle">
-          <p>박스오피스 순위</p>
+          <p>{listTitle}</p>
         </div>
-        <section className="mainMovie">
+        <button
+          disabled={index === 0 && 'disabled'} //초기값일때 이전버튼 비활성화
```

Comment:
disabled 값은 true false로 들어가야되는걸로 보입니다!
"disabled" 라는 스트링을 값으로 넣어줄 필요는 없을 것 같네요(동작은 할 거에요 값이 있는 스트링은 true 처리되니까)
```suggestion
          disabled={!index} //초기값일때 이전버튼 비활성화
```
이정도로만 해줘도 충분해보입니다!
index가 0일때 비활성화 되는 걸 좀 더 가독성 측면에서 좀 더 명시적으로 표현하고 싶다면
```suggestion
          disabled={index === 0} //초기값일때 이전버튼 비활성화
```
정도가 좋아보이네요!

별개로 index라는 변수명이 현재 활성화 된 slide 번호 라는 측면이 잘 들어나지는 않아보이는 변수명 같네요!
고민하고 수정해주시면 좋겠네요!

Reply:
변수명들 모두 수정했습니다. 조건 또한 명시적으로 바꿔주었습니다! '!'는 생각도 못했는데 좋은 방법 같네요